packages/kit/src/runtime/client/client.js

```diff
@@ -259,6 +269,7 @@ export function create_client({ target, session, base, trailing_slash }) {
 			const { details } = opts;
 			const change = details.replaceState ? 0 : 1;
 			details.state[INDEX_KEY] = current_history_index += change;
+			details.state[RESTORATION_KEY] = current_position_index = Math.random().toString(32).slice(2);
```

Comment:
I'm not sure whether I should put this in `if(!details.replaceState)` contitional

Reply:
I did put in inside conditional